tests/functional/test_files.py

```diff
@@ -101,14 +101,28 @@ def test_download_file_credentials(self):
         file_info = self.files.detail(file_id, federation_token=True)
         with self.subTest():
             self.assertEqual(file_id, file_info['id'])
-        with self.subTest():
-            self.assertTrue('credentials' in file_info)
-        with self.subTest():
-            self.assertTrue('AccessKeyId' in file_info['credentials'])
-        with self.subTest():
-            self.assertTrue('SecretAccessKey' in file_info['credentials'])
-        with self.subTest():
-            self.assertTrue('SessionToken' in file_info['credentials'])
+        if file_info['provider'] == 'aws':
```

Comment:
this is not good - it can happen that there are no assertions executed at all

Reply:
you mean if there is no provider?  that's not really possible

Reply:
I wanted to make the tests backend agnostic,  but there's probably a better way, I'll try to come up with something.

Reply:
or a different one then `aws` or `azure`

Reply:
maybe you can only assert that the the value is one of the allowed values and assert that credentials are present? the rest is the job of the API itself

Reply:
ah, yeah,  but it would be nice if the tests break on any BC breaks in the api